torch/csrc/autograd/python_variable.cpp

```diff
@@ -684,6 +684,36 @@ static PyObject* THPVariable_fix_weakref(PyObject* self, PyObject* noargs) {
   Py_RETURN_NONE;
 }
 
+static PyObject* THPVariable_view_func(PyObject* self_, PyObject* arg) {
+  HANDLE_TH_ERRORS
+  const auto& self = THPVariable_Unpack(self_);
+  TORCH_CHECK(
+      THPVariable_Check(arg),
+      "_view_func expect a single argument that is a Tensor");
+  const auto& new_base = THPVariable_Unpack(arg);
+
+  // Ensure that self is indeed a backward differentiable view
+  auto diff_view_meta = torch::autograd::impl::get_view_autograd_meta(self);
+  TORCH_CHECK(
+      diff_view_meta && diff_view_meta->has_bw_view(),
+      "_view_func can only be called on "
+      "a Tensor that is a backward differentiable view.");
+  const auto& view_info = diff_view_meta->get_backward_view();
+  // Ensure that the newly provided base is similar to the original base
```

Comment:
The reason we have this check is because to avoid situations where a view is no longer valid on the new base? For example, flattening a non-contiguous tensor. 

Certain attributes don't necessarily have to match though. For example, if storage numel of the new base is larger, any view on the original base should also be OK on the new base.

I guess we can start out with just making everything match though, since it is easier lol

Reply:
Hopefully AOTAutograd is OK with these restrictions?

Reply:
This is kind of a dangerous API so I prefer to be too strict than not enough.
But also, since this might call as_strided, you do want the metadata to match very closely. Otherwise you might get very weird result!